SU2_CFD/src/variables/CMeshVariable.cpp

```diff
@@ -48,19 +48,8 @@ CMeshVariable::CMeshVariable(unsigned long npoint, unsigned long ndim, CConfig *
   }
 }
 
-void CMeshVariable::Register_MeshCoord(bool input) {
-  if (input) {
-    SU2_OMP_FOR_STAT(roundUpDiv(nPoint,omp_get_num_threads()))
-    for (unsigned long iPoint = 0; iPoint < nPoint; iPoint++)
-      for (unsigned long iDim = 0; iDim < nDim; iDim++)
-        AD::RegisterInput(Mesh_Coord(iPoint,iDim));
-    END_SU2_OMP_FOR
-  }
-  else {
-    SU2_OMP_FOR_STAT(roundUpDiv(nPoint,omp_get_num_threads()))
-    for (unsigned long iPoint = 0; iPoint < nPoint; iPoint++)
-      for (unsigned long iDim = 0; iDim < nDim; iDim++)
-        AD::RegisterOutput(Mesh_Coord(iPoint,iDim));
-    END_SU2_OMP_FOR
-  }
+void CMeshVariable::Register_MeshCoord() {
+  for (unsigned long iPoint = 0; iPoint < nPoint; iPoint++)
+    for (unsigned long iDim = 0; iDim < nDim; iDim++)
+      AD::RegisterInput(Mesh_Coord(iPoint,iDim));
```

Comment:
Just being curious, how come that neither indices of updated mesh coordinates nor displacements (see above) are needed? I thought their adjoints must be initialised through at least one of them?

Reply:
They are only used as inputs, but in #1284 I also changed this to store the input indices (just for consistency).

Reply:
Ok. :) Right, I didn't think of the discrete adjoint FEA solver for a second where the adjoint sweep goes through. I think I've worked out the precise initialize/extract logic for FSI couplings in one of your PR's one day but it's a while ago already...